packages/monaco/src/browser/monaco-command.ts

```diff
@@ -129,12 +135,11 @@ export class MonacoEditorCommandHandlers implements CommandContribution {
      * and execute them using the instantiation service of the current editor.
      */
     protected registerMonacoCommands(): void {
-        const editorRegistry = monaco.editorExtensions.EditorExtensionsRegistry;
-        const editorActions = new Map(editorRegistry.getEditorActions().map(({ id, label }) => [id, label]));
+        const editorActions = new Map(EditorExtensionsRegistry.getEditorActions().map(({ id, label }) => [id, label]));
 
-        const { codeEditorService, textModelService, contextKeyService } = this;
-        const [, globalInstantiationService] = monaco.services.StaticServices.init({ codeEditorService, textModelService, contextKeyService });
-        const monacoCommands = monaco.commands.CommandsRegistry.getCommands();
+        const { codeEditorService } = this;
+        const globalInstantiationService = StandaloneServices.initialize({});
```

Comment:
Only overrides passed in the first call to `initialize` will be applied to the `IInstantiationService` that is returned from all subsequent calls to `initialize`, but all calls to `initialize` will return the same `InstantiationService`, so calling `initialize` later is still an effective way to get access to it.

Reply:
I don't like this one: it means we're order dependent in the initialization. Is there not other way to obtain the services?

Reply:
The short answer to whether there's another way is 'no', but there is a longer answer. Basically, the InstantiationService is the equivalent of our `inversify` containers, and the one returned from `StandaloneServices.initialize()` is the equivalent of our global container. Just like with inversify, it's possible to create sub-containers, and [I've adjusted](https://github.com/eclipse-theia/theia/pull/10736/commits/cec56968bad3cf285217eab152daf09b0f3858f6) some of our code to use a sub-container strategy for some of the overrides that we were passing to editors.

We could write a heavier wrapper around that system, but at the moment it's a bit opaque to me: I've tried to get as close to the old behavior as possible, given the new 'initialize-once-use-many' system, but I'm not sure in every case (or in many cases) why a particular override is being passed at a particular point: whether it's actually different from what's available in the global instantiation service and whether we're creating individual overrides in cases where we could use global overrides. And previously, since the override object was essentially untyped (`{[key: string]: any}`), it wasn't even clear whether the overrides were actually being used at all - I suspect more than one was just being ignored.

And now that I've bloviated, a bit, let me address your actual concern: at the moment, the intended 'first' initialization is taking place in `MonacoFrontendApplicationContribution.initialize`, which is as early in the application lifecycle as we can get. Everything passed in there is available from the global `inversify` container, so we could move it into a `postConstruct` method and get a little earlier, or move it into `monaco-frontend-module` to (try to) guarantee that the first call to `initialize` happens before the normal application lifecycle kicks off - but that would make it less amenable to extension.

Reply:
And now that I've bloviated even more, how about this idea: we create a symbol/type pair for global overrides that should be applied on initialization, and apply those in the `postConstruct` of the `MonacoFrontendApplicationContribution`? that would make it pretty straightforward to change the set of overrides.

Reply:
It would likely be the latter: every call would succeed, but at least some of the services we expected to be active in Monaco might not be. In general, the right way to handle it would likely be never to call `.initialize()` apart from intended initialization. There are at least a couple of cases where we've decorated a Monaco class as injectable and then given ourselves access to it that way, and that's less risky. Since it's imported from Monaco directly in the places its used, it's hard to stop people accessing it, but we could monkey patch it at import time to be either something like this:

```typescript
const originalInitialize = StandaloneServices.initialize
const ourInitialize = (overrides) => {
    const merged = {...overrides, ...ourIntendedOverrides};
    return originalInitialize(merged);
};
StandaloneServices.initialize = ourInitialize;
```
Or something like this:
```typescript
let initialized = false;
const originalInitialized = StandaloneServices.initialize
const ourInitialize = (overrides) => {
    if (!initialized) {
        initialized = true;
        console.log('Initializing Monaco services with specified overrides:', overrides, '. Overrides will be ignored for all subsequent calls to this function.');
    }
    return originalInitialize(overrides);
}
StandaloneServices.initialize = ourInitialize;
```

The former would guarantee that our intended overrides (however we wanted to retrieved them) got applied, and the latter would at least give the astute log reader a hint about where things might have gone wrong. What do you think of those options, @tsmaeder, or do you have something else in mind?